modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -8531,6 +8531,45 @@ final boolean traverse(Direction dir, TraversalMethod method) {
         return getScene().traverse(this, dir, method);
     }
 
+    /**
+     * Requests to move the focus from this {@code Node} in the specified direction.
+     * The {@code Node} serves as a reference point and does not have to be focused or focusable.
+     * A successful traversal results in a new {@code Node} being focused.
+     * <p>
+     * This method is expected to be called in response to a {@code KeyEvent}; therefore the {@code Node}
+     * receiving focus will have the {@link #focusVisibleProperty() focusVisible} property set.
+     *
+     * @param direction the direction of focus traversal
+     * @return {@code true} if traversal was successful
+     * @since 24
+     */
+    public final boolean requestFocusTraversal(TraversalDirection direction) {
+        Direction d;
+        switch (direction) {
+        case DOWN:
```

Comment:
Have you considered moving this code to the `Direction` enum, for example like this:
```java
enum Direction {
    ...
    public static Direction of(TraversalDirection direction) {
        return switch (direction) {
            case DOWN -> DOWN;
            case LEFT -> LEFT;
            case NEXT -> NEXT;
            case PREVIOUS -> PREVIOUS;
            case RIGHT -> RIGHT;
            case UP -> UP;
        }
    }
}
```

This way we won't clutter up the already very long `Node` file with simple conversions, and the co-location of the conversion with the actual enum makes it easier to maintain the code as well.